api/filters/replacement/replacement.go

```diff
@@ -93,10 +119,28 @@ func getReplacement(nodes []*yaml.RNode, r *types.Replacement) (*yaml.RNode, err
 	if err != nil {
 		return nil, err
 	}
-	// TODO (#3492): Use the field options to refine interpretation of the field
+	if !rn.IsNilOrEmpty() && rn.YNode().Kind == yaml.ScalarNode {
+		return getRefinedValue(r.Source.Options, rn)
+	}
 	return rn, nil
 }
 
+func getRefinedValue(options *types.FieldOptions, rn *yaml.RNode) (*yaml.RNode, error) {
+	if options == nil || options.Delimiter == "" {
+		return rn, nil
+	}
+	value := strings.Split(yaml.GetValue(rn), options.Delimiter)
+	var index int
+	if options.Index > len(value) || options.Index < 0 { // take the last element
```

Comment:
This also seems like invalid configuration to me that should result in an error. It makes sense to provide a way to target the last element regardless of how many there are, but I'd say that should be done with a specific value, i.e. `-1`, or for negative indexes to be supported in general. But I feel like `> current length` and `< 0` both meaning "last" could get confusing, especially given that current length can change independently of the Kustomization this is written in.

Reply:
I have updated it to throw an error if the source index is out of bounds

Reply:
It's a good idea to be relaxed about inputs (and strict on outputs).

However, I cannot  think of a no-surprises default behavior for an out-of-range index after the split when trying to get the source value.

So - seems that it should error (fail build).  Meaning that a split resulting in an empty array is always an error.